comp/core/flare/flare.go

```diff
@@ -140,14 +141,19 @@ func (f *flare) createAndReturnFlarePath(w http.ResponseWriter, r *http.Request)
 		}
 	}
 
+	queryProviderTimeout := r.URL.Query().Get("provider_timeout")
+	providerTimeoutSeconds, err := strconv.Atoi(queryProviderTimeout)
+	if err != nil || providerTimeoutSeconds <= 0 {
+		providerTimeoutSeconds = f.config.GetInt("flare_provider_timeout")
+	}
```

Comment:
Since we call `Create` below on line `177` we could avoid assigning `f.config.GetInt("flare_provider_timeout")` twice.


```go
queryProviderTimeout := r.URL.Query().Get("provider_timeout")
providerTimeoutSeconds, err := strconv.Atoi(queryProviderTimeout)
if err != nil {
	providerTimeoutSeconds = 0
}
```

Then in line `178` we perform the check just once 😄